data/modules/Scoop/Scoop.lua

```diff
@@ -0,0 +1,837 @@
+	local ship
+	local police = {}
+	local shipdef = ShipDef[Game.system.faction.policeShip]
+
+	for i = 1, 2 do
+		ship = Space.SpawnShipDocked(shipdef.id, station)
+		ship:SetLabel(lc.POLICE)
+		ship:AddEquip(Equipment.laser.pulsecannon_1mw)
+		table.insert(police, ship)
+		if station.type == "STARPORT_SURFACE" then
+			ship:AIEnterLowOrbit(Space.GetBody(station:GetSystemBody().parent.index))
+		end
+	end
+
+	Timer:CallAt(Game.time + 5, function ()
+		for _, s in pairs(police) do
+			s:AIKill(Game.player)
+		end
+	end)
+
+	return police
+end
+
+-- Returns a random system close to the players location
+local nearbySystem = function ()
+	local dist = 5
+	local systems = {}
+
+	while #systems < 1 do
+		systems = Game.system:GetNearbySystems(dist)
+		dist = dist + 5
+	end
+
+	return systems[Engine.rand:Integer(1, #systems)].path
+end
+
+-- Create a pirate ship
+local spawnPirate = function (star, pirate_label)
+	local shipdefs = utils.build_array(utils.filter(
+		function (k, def)
+			return def.tag == "SHIP" and def.hyperdriveClass > 0 and def.equipSlotCapacity["scoop"] > 0
+		end,
+		pairs(ShipDef)))
+	local shipdef = shipdefs[Engine.rand:Integer(1, #shipdefs)]
+
+	local radius = star:GetSystemBody().radius
+	local min, max
+	if star:GetSystemBody().type == "WHITE_DWARF" then
+		min = radius * 30
+		max = radius * 40
+	else
+		min = radius * 3.5
+		max = radius * 4.5
+	end
+
+	local ship = Space.SpawnShipOrbit(shipdef.id, Space.GetBody(star:GetSystemBody().index), min, max)
+
+	ship:SetLabel(pirate_label)
+	ship:AddEquip(Equipment.hyperspace["hyperdrive_" .. shipdef.hyperdriveClass])
+	ship:AddEquip(Equipment.laser.pulsecannon_2mw)
+	ship:AddEquip(Equipment.misc.shield_generator)
+	local c = math.min(ship.totalCargo - ship.usedCargo - 1, 10)
+	ship:AddEquip(Equipment.cargo.hydrogen, c)
+
+	return ship
+end
+
+local removeMission = function (mission, ref)
+	local oldReputation = Character.persistent.player.reputation
+	local sender = mission.pirate and mission.pirate_label or mission.client.name
+
+	if mission.status == "COMPLETED" then
+		Character.persistent.player.reputation = oldReputation + mission_reputation
+		Game.player:AddMoney(mission.reward)
+		Comms.ImportantMessage(l["SUCCESS_MSG_" .. mission.id], sender)
+	elseif mission.status == "FAILED" then
+		Character.persistent.player.reputation = oldReputation - mission_reputation
+		Comms.ImportantMessage(l["FAILURE_MSG_" .. mission.id], sender)
+	end
+	Event.Queue("onReputationChanged", oldReputation, Character.persistent.player.killcount,
+		Character.persistent.player.reputation, Character.persistent.player.killcount)
+
+	if ref == nil then
+		for r, m in pairs(missions) do
+			if m == mission then ref = r break end
+		end
+	end
+	mission:Remove()
+	missions[ref] = nil
+end
+
+-- Cargo transfer to a ship
+local transferCargo = function (mission)
+	Timer:CallEvery(9, function ()
+		if not mission.pirate then return true end
+
+		if Game.player:DistanceTo(mission.pirate) <= 50 then
```

Comment:
This function ideally needs a Comms message when the player closes to within ~5km informing the player that they need to fly close enough to transfer cargo. It's otherwise non-obvious how the player should complete the mission, and even after skimming through the code I found myself having to double-check the completion condition.

Additionally, a 50m distance can fail for some ship types (dsminer, malabar) especially if the player is also in a larger ship. The *proper* solution here is to add the player and the target's clip radius to the distance, but that's not currently exposed to Lua. Failing that, increasing the distance to e.g. 100m should work correctly for all ships.